src/main/java/org/dependencytrack/persistence/H2WebConsoleInitializer.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * This file is part of Dependency-Track.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) Steve Springett. All Rights Reserved.
+ */
+
+package org.dependencytrack.persistence;
+
+import alpine.Config;
+import alpine.common.logging.Logger;
+import org.dependencytrack.common.ConfigKey;
+import org.h2.server.web.WebServlet;
+
+import javax.servlet.ServletContext;
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import javax.servlet.ServletRegistration;
+import java.util.Map;
+
+public class H2WebConsoleInitializer implements ServletContextListener {
+    private static final Logger LOGGER = Logger.getLogger(H2WebConsoleInitializer.class);
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void contextInitialized(final ServletContextEvent event) {
+        Config configuration = Config.getInstance();
+        String databaseMode = configuration.getProperty(Config.AlpineKey.DATABASE_MODE);
+        String databaseDriver = configuration.getProperty(Config.AlpineKey.DATABASE_DRIVER);
+        Boolean h2ConsoleEnabled = configuration.getPropertyAsBoolean(ConfigKey.H2_CONSOLE_ENABLED);
+        // Misconfiguration check, if external database is used, no need to pointlessly expose the H2 console
+        if ("external".equals(databaseMode) || !org.h2.Driver.class.getName().equals(databaseDriver) || !h2ConsoleEnabled) {
+            LOGGER.debug("H2 web console will not be initialized since either database mode is external or database driver is not H2 or the console is simply disabled !");
+            LOGGER.debug("Database mode : "+databaseMode);
+            LOGGER.debug("Database driver : "+databaseDriver);
+            LOGGER.debug("H2 web console enabled : "+h2ConsoleEnabled);
+            return;
+        }
```

Comment:
As exposing the console is a security issue in itself, it would be good to log a warning that it is enabled.

I do hope no one is using H2 in production, but we should make it clear that the console is only intended for local testing.

Reply:
Done.